python/xoscar/collective/core.py

```diff
@@ -133,10 +182,16 @@ def allreduce(
         algorithm: AllReduceAlgorithm = AllReduceAlgorithm.RING,
         tag: Optional[int] = 0,
         pg_name: str = "default",
+        stream: Optional[Any] = None,
```

Comment:
This API design does not cover the two backends. e.g. the stream is only for nccl backend, algorithm is only for the gloo backend.